07_memory/user_mem/user_mem.c

```diff
@@ -0,0 +1,191 @@
+#include <stdio.h>
+#include <time.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <malloc.h>
+#include <stdbool.h>
+#include <sys/time.h>
+#include <sys/resource.h>
+#include <unistd.h>
+
+#define NSEC_IN_SEC     1e9
+#define MAX_BUF_SIZE    64
+#define NUM_OF_ITER 	10
+
+#define POW2(x)     (1LL << (x))
+
+#define MIN(x, y) x > y ? y : x
+#define MAX(x, y) x > y ? x : y
+
+uint64_t get_diff(struct timespec t1, struct timespec t2)
+{
+	uint64_t sec_diff;
+	uint64_t nsec_diff;
+
+	sec_diff = t1.tv_sec - t2.tv_sec;
+	nsec_diff = t1.tv_nsec - t2.tv_nsec;
+
+	if (nsec_diff < 0) {
+		nsec_diff += NSEC_IN_SEC;
+		sec_diff--;
+	}
```

Comment:
Is this IF block required?

Reply:
Yes, it's necessary for cases when _t1_ and _t2_ are differ for 1 or more seconds.